enhancements/sig-architecture/20-taint-toleration/README.md

```diff
@@ -0,0 +1,318 @@
+   // Valid operators are Exists and Equal. Defaults to Equal.
+   // Exists is equivalent to wildcard for value, so that a placement can
+   // tolerate all taints of a particular category.
+   // +optional
+   Operator TolerationOperator `json:"operator,omitempty"`
+   // Value is the taint value the toleration matches to.
+   // If the operator is Exists, the value should be empty, otherwise just a regular string.
+   // +optional
+   Value string `json:"value,omitempty"`
+   // TolerationSeconds represents the period of time the toleration (which must only be effective under the Evict Senario, otherwise this field should be ignored) tolerates the taint. By default,
+   // it is not set, which means tolerate the taint forever (do not evict). 
+   // The start time of counting the TolerationSeconds should be the TimeAdded in Taint, not the cluster scheduled time or TolerationSeconds added time.
+   // +optional
+   TolerationSeconds *int64 `json:"tolerationSeconds,omitempty"`
+}
+```
+
+### Placement repo
+
+#### Four steps of current schedule:
+
+* Filter: Get clusters satisfying all the requirements from all the clusters. 
+
+* Score: Sort by certain criteria.
+
+* Select: Only select the number of clusters we need, not all feasible clusters.
+
+* Bind: merge the cluster decisions into placementdecisions
+
+#### Changes:
+
+* Current placement process is implemented by plugins. To add taint-toleration, we need to add a new plugin which check taint-toleration matching after the filter pipline in the schedule function. Specifically, we do following matching checking in this new plugin:
+  * for every taint of all the taints on a managed cluster
+  * go through every toleration on the placement
+  * check if there is a match
+* By the end of the schedule process, we check all tolerated taints to get the first expiration time of the cluster.
+* `scheduleResult` should be updated by adding `requeue` and `delay`, which represent whether requeue need to be triggered in the controller and after how many seconds it would be triggerd. 
+* We also add a requeue process in the controller to evict the cluster after this `delay` time. `delay` time is the first expiration time of tolerated taints, and it would be computed as: toleration.TolerationSeconds - (current time - taint.TimeAdded), and the first expiration time is the minimum among all expiration times.
+* Update the plugin interface, which can return the `requeueAfter` time for each plugin.
+
+### Register repo
+
+#### Add new controller:
+
+Based on what we have in current Placement APIs, we add a new controller which can map the state of unhealthy and unreachable managed clusters to corresponding taints `open-clustet-management.io/unhealthy` and `open-cluster-management.io/unreachable`, so we can leverage the Taint-Toleration to support[ filtering unhealthy/not-reporting clusters (issue #48)](https://github.com/open-cluster-management-io/community/issues/48).
```

Comment:
why wouldn't we have the same controller set both values?

Reply:
I currently don't change the controller part.

Reply:
Makes sense.